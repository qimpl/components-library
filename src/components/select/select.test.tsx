import React from 'react';

import { render } from '@testing-library/react';

import Select from 'components/select';

const defaultOption = {
  id: 'disabled-option',
  value: '',
  label: 'Select',
};

const options = [
  {
    id: '1',
    value: '1',
    label: '1st value',
  },
  {
    id: '2',
    value: '2',
    label: '2nd value',
  },
  {
    id: '3',
    value: '3',
    label: '3rd value',
  },
];

test('Default select', () => {
  expect(
    render(
      <Select
        id='my-select'
        label='Choose'
        onChange={() => 'changed'}
        onBlur={() => 'blurred'}
        value=''
        defaultOption={defaultOption}
        options={options}
      />
    )
  ).toMatchSnapshot();
});

test('Pre-selected value', () => {
  expect(
    render(
      <Select
        id='my-select'
        label='Choose'
        onChange={() => 'changed'}
        onBlur={() => 'blurred'}
        value='2'
        defaultOption={defaultOption}
        options={options}
      />
    )
  ).toMatchSnapshot();
});

test('Select with errror', () => {
  expect(
    render(
      <Select
        id='error-select'
        label='Choose'
        onChange={() => 'changed'}
        onBlur={() => 'blurred'}
        value=''
        error='Select a value'
        defaultOption={defaultOption}
        options={options}
      />
    )
  ).toMatchSnapshot();
});

test('Select without label', () => {
  expect(
    render(
      <Select
        id='select-no-label'
        onChange={() => 'changed'}
        onBlur={() => 'blurred'}
        value=''
        defaultOption={defaultOption}
        options={options}
      />
    )
  ).toMatchSnapshot();
});
