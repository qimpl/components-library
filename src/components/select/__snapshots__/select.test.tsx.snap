// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default select 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ gkORIc"
      >
        <label
          for="my-select"
        >
          Choose
        </label>
        <select
          id="my-select"
          name="my-select"
        >
          <option
            disabled=""
            value=""
          >
            Select
          </option>
          <option
            value="1"
          >
            1st value
          </option>
          <option
            value="2"
          >
            2nd value
          </option>
          <option
            value="3"
          >
            3rd value
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ gkORIc"
    >
      <label
        for="my-select"
      >
        Choose
      </label>
      <select
        id="my-select"
        name="my-select"
      >
        <option
          disabled=""
          value=""
        >
          Select
        </option>
        <option
          value="1"
        >
          1st value
        </option>
        <option
          value="2"
        >
          2nd value
        </option>
        <option
          value="3"
        >
          3rd value
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Pre-selected value 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ gkORIc"
      >
        <label
          for="my-select"
        >
          Choose
        </label>
        <select
          id="my-select"
          name="my-select"
        >
          <option
            disabled=""
            value=""
          >
            Select
          </option>
          <option
            value="1"
          >
            1st value
          </option>
          <option
            value="2"
          >
            2nd value
          </option>
          <option
            value="3"
          >
            3rd value
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ gkORIc"
    >
      <label
        for="my-select"
      >
        Choose
      </label>
      <select
        id="my-select"
        name="my-select"
      >
        <option
          disabled=""
          value=""
        >
          Select
        </option>
        <option
          value="1"
        >
          1st value
        </option>
        <option
          value="2"
        >
          2nd value
        </option>
        <option
          value="3"
        >
          3rd value
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Select with errror 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ hVVFSP"
      >
        <label
          for="error-select"
        >
          Choose
        </label>
        <select
          id="error-select"
          name="error-select"
        >
          <option
            disabled=""
            value=""
          >
            Select
          </option>
          <option
            value="1"
          >
            1st value
          </option>
          <option
            value="2"
          >
            2nd value
          </option>
          <option
            value="3"
          >
            3rd value
          </option>
        </select>
        <span>
          Select a value
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ hVVFSP"
    >
      <label
        for="error-select"
      >
        Choose
      </label>
      <select
        id="error-select"
        name="error-select"
      >
        <option
          disabled=""
          value=""
        >
          Select
        </option>
        <option
          value="1"
        >
          1st value
        </option>
        <option
          value="2"
        >
          2nd value
        </option>
        <option
          value="3"
        >
          3rd value
        </option>
      </select>
      <span>
        Select a value
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Select without label 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ gkORIc"
      >
        <select
          id="select-no-label"
          name="select-no-label"
        >
          <option
            disabled=""
            value=""
          >
            Select
          </option>
          <option
            value="1"
          >
            1st value
          </option>
          <option
            value="2"
          >
            2nd value
          </option>
          <option
            value="3"
          >
            3rd value
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ gkORIc"
    >
      <select
        id="select-no-label"
        name="select-no-label"
      >
        <option
          disabled=""
          value=""
        >
          Select
        </option>
        <option
          value="1"
        >
          1st value
        </option>
        <option
          value="2"
        >
          2nd value
        </option>
        <option
          value="3"
        >
          3rd value
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
